clang/lib/Sema/SemaARM.cpp

```diff
@@ -1328,4 +1328,57 @@ void SemaARM::handleInterruptAttr(Decl *D, const ParsedAttr &AL) {
                  ARMInterruptAttr(getASTContext(), AL, Kind));
 }
 
+// Check if the function definition uses any AArch64 SME features without
+// having the '+sme' feature enabled and warn user if sme locally streaming
+// function returns or uses arguments with VL-based types.
+void SemaARM::CheckSMEFunctionDefAttributes(const FunctionDecl *FD) {
```

Comment:
Should this be a `static` member function instead? (would have to take the `ASTContext` as parameter)

Reply:
I've changed this to `static` and added `Sema` as a parameter, so that the function can still use `Diag()` for emitting the diagnostics.

Reply:
It's a bit inconsistent with everything else, the intent is to use `ARM().CheckSMEFunctionDefAttributes(FD)` 
(The reason we did that was to split Sema into multiple classes/files)

@Endilll

Reply:
It seems like an ARM-specific function, so you correctly placed it in `SemaARM`, and it's referenced in SemaDecl and SemaDeclCXX, so it can't be static. Corentin is correct that `ARM().CheckSMEFunctionDefAttributes(FD)` is the way to use it from outside of SemaARM.

The current state of the changes seems correct to me.

Reply:
Thanks for your feedback, I didn't realise that there was already precedent for using `ARM().Check...`